### ISSUE TITLE
Chore: Run Tests on `GitHub`  Actions

### ISSUE THREAD

**Issue by nelsonic** [dwyl/flutter-counter-example#6]
# Story

As an engineer building a `Flutter` App, 🧑‍💻 
I want to run tests whenever new code is committed ⬆️ 
So that I know everything is working as expected. ✅ 

# Todo

+ [ ] Figure out how to run the test for this basic counter on **`GitHub` Actions**. 🔍 

> **Note**: the https://github.com/dwyl/flutter-todo-list-tutorial/pull/28 already has CI/CD running via: [`.github/workflows/ci.yml`](https://github.com/dwyl/flutter-todo-list-tutorial/blob/master/.github/workflows/ci.yml)  👀 
this will be a good starting point for your quest. 🚀 
The versions of `Java` and `Flutter` will need to be updated but the rest should be useful. 🤞 

Please leave as many comments as you can on this issue in your quest to get the tests running on **`GitHub` Actions**. 💬 
This is a bit of an "unknown" for us because we have limited `Flutter` experience so this needs to be very well documented. 🙏

**Comment by nelsonic** [comment]
Given that we already have CI/CD working on one of our Flutter repos: https://github.com/dwyl/flutter-todo-list-tutorial/pull/28 ... 
I _estimate_ that this might take less time ... but feel free to add your own time estimate label. 👌

**Comment by LuchoTurtle** [comment]
@nelsonic this might seem like a dumb question but how am I meant to test if a Github Action pipeline works from a PR? Don't I need to push it to `main` to get it working? :/

**Comment by nelsonic** [comment]
@LuchoTurtle good question. Not "dumb" at all. Always ask questions even if you _think_ they might be _obvious_. 👌 

GitHub Actions do indeed work properly once the `.github/` directory is merged into `main`.
We sometimes create a separate micro-PR just to add GitHub Actions + @dependabot 
e.g: https://github.com/dwyl/learn-tdd/pull/139 or https://github.com/dwyl/javascript-todo-list-tutorial/pull/35

And then once GitHub CI is working on the `main` branch,a the subsequent PRs everything works as expected. ✅

**Comment by nelsonic** [comment]
Keeping open until you've confirmed it's working. 🙏

**Comment by LuchoTurtle** [comment]
Thanks. I realised this is an older repo, so I have to change the config file for `master` branch instead of `main` 🤦

**Comment by nelsonic** [comment]
@LuchoTurtle nah, let's just rename the `default` branch from `master` to `main`:
https://github.com/dwyl/flutter-counter-example/settings/branches

<img width="1049" alt="image" src="https://user-images.githubusercontent.com/194400/202155773-aea2b9ad-1405-41db-9999-48f5f722d4dd.png">

Once you rename the branch the CI will work. :ok_hand:

**Comment by nelsonic** [comment]
<img width="1049" alt="image" src="https://user-images.githubusercontent.com/194400/202155517-ae3e54fa-a2ba-4c21-81ee-abd2ab667470.png">

Once you rename the branch the CI will work. :ok_hand:

**Comment by nelsonic** [comment]
Re-run the job: https://github.com/dwyl/flutter-counter-example/actions/runs/3478443249

**Comment by nelsonic** [comment]
https://github.com/dwyl/flutter-counter-example/actions/runs/3478443249/jobs/5815891718
<img width="1138" alt="image" src="https://user-images.githubusercontent.com/194400/202156140-0c393ec3-c0e8-446b-be87-af2c7d318037.png">

# Error: unable to find release for `3.3.8` 🤷‍♂️ 

I'm sure you're already on the case. 🔍

**Comment by LuchoTurtle** [comment]
@nelsonic it _should be fixed_ but the script uses a Flutter version that is current, unlike the current project on the `main` branch. However, it *should* work when merging the #7 PR, as it uses the latest Flutter project boilerplate and dependencies.

**Comment by nelsonic** [comment]
Agreed. let's get the PR merged. 👌

**Comment by nelsonic** [comment]
Just FYI: the PR I created https://github.com/dwyl/protect-main-branch/pull/9/files adds GitHub Actions CI and you can see the Codecov comment in the PR: https://github.com/dwyl/protect-main-branch/pull/9#issuecomment-1317099900 so sometimes it works.

**Comment by LuchoTurtle** [comment]
As stated in https://github.com/dwyl/flutter-counter-example/pull/7#issuecomment-1332713883, this issue should be resolved once the PR is merged.

**Comment by nelsonic** [comment]
PR merged. :shipit:

**Comment by LuchoTurtle** [comment]
> PR merged. :shipit:

PR has not been merged yet https://github.com/dwyl/flutter-counter-example/pull/7

**Comment by nelsonic** [comment]
Check again. 😉 :shipit:
